src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift

```diff
@@ -36,120 +37,145 @@ final class PPAnalyticsSubmitter: PPAnalyticsSubmitting {
 		completion: ((Result<Void, PPASError>) -> Void)? = nil
 	) {
+		Log.info("Analytics submission was triggered. Checking now if we can submit...", log: .ppa)
+		
+		// Check if a submission is already in progress
+		guard submissionState == .readyForSubmission else {
+			Log.warning("Analytics submission abord due to submission is already in progress", log: .ppa)
+			completion?(.failure(.submissionInProgress))
+			return
+		}
+		
+		submissionState = .submissionInProgress
 		
 		// Check if user has given his consent to collect data
 		if userDeclinedAnalyticsCollectionConsent {
 			Log.warning("Analytics submission abord due to missing users consent", log: .ppa)
+			submissionState = .readyForSubmission
 			completion?(.failure(.userConsentError))
 			return
 		}
 		
-		if cancellable == nil {
-			cancellable = configurationProvider.appConfiguration().sink { [weak self] configuration in
-				
-				guard let self = self else {
-					Log.warning("Analytics submission abord due fail at creating strong self", log: .ppa)
-					completion?(.failure(.generalError))
-					return
-				}
-				
-				// Check configuration parameter
-				if Double.random(in: 0...1) > configuration.privacyPreservingAnalyticsParameters.common.probabilityToSubmit {
-					Log.warning("Analytics submission abord due to randomness", log: .ppa)
-					completion?(.failure(.probibilityError))
-					return
-				}
-				
-				// Last submission check
-				if self.submissionWithinLast23Hours {
-					Log.warning("Analytics submission abord due to submission last 23 hours", log: .ppa)
-					completion?(.failure(.submission23hoursError))
-					return
-				}
-				
-				// Onboarding check
-				if self.onboardingCompletedWithinLast24Hours {
-					Log.warning("Analytics submission abord due to onboarding completed last 24 hours", log: .ppa)
-					completion?(.failure(.onboardingError))
-					return
-				}
-				
-				// App Reset check
-				if self.appResetWithinLast24Hours {
-					Log.warning("Analytics submission abord due to app resetted last 24 hours", log: .ppa)
-					completion?(.failure(.appResetError))
-					return
-				}
-				
-				self.hoursSinceTestResultToSubmitKeySubmissionMetadata = configuration.privacyPreservingAnalyticsParameters.common.hoursSinceTestResultToSubmitKeySubmissionMetadata
-				self.hoursSinceTestRegistrationToSubmitTestResultMetadata = configuration.privacyPreservingAnalyticsParameters.common.hoursSinceTestRegistrationToSubmitTestResultMetadata
-				self.probabilityToSubmitExposureWindows = configuration.privacyPreservingAnalyticsParameters.common.probabilityToSubmitExposureWindows
-
-				if let token = ppacToken {
-					// Submit analytics data with injected ppac token
-					self.submitData(with: token, completion: completion)
-				} else {
-					self.generatePPACAndSubmitData(completion: completion)
-				}
+		// Sink on the app configuration if something has changed. But do this in background and wait for the result before continue.
```

Comment:
I'm wondering because the request to submit should be async, why is it needed to operate on the background here?
Can you write in the comment, why it's needed to call this on the background?

Reply:
We had already an issue that the main thread was blocked by the analytics submission. So I thought it would be better to do this in the background that the main is never blocked because of submitting the analytics data, what the user is not needed for a working app.